### PR TITLE
ROX-14701: Document how to download roxctl for IBM P&Z

### PR DESCRIPTION
**Pull request by msugakov**
Version(s):
* Merge to `rhacs-docs`
* Cherry pick to `rhacs-docs-4.3`

Issue: https://issues.redhat.com/browse/ROX-14701

Link to docs preview: https://65916--docspreview.netlify.app/openshift-acs/latest/cli/installing-the-roxctl-cli

QE review:
- [ ] QE has approved this change.

Additional information:
I recommend reviewing this change per commit as I tried providing meaningful commit messages in each.
This complements https://github.com/stackrox/stackrox/pull/8011 by extending documentation.

Testing done:
I created a small script with the similar content and ran it against `zsh`, `bash` and `dash` to make sure commands are valid.

```bash
#!/usr/bin/env zsh

set -eux

arch="$(uname -m | sed "s/x86_64//")"; arch="${arch:+-$arch}"
#arch="$(echo "s390x" | sed "s/x86_64//")"; arch="${arch:+-$arch}"

curl -f -o roxctl "https://mirror.openshift.com/pub/rhacs/assets/4.2.0/bin/Linux/roxctl${arch}"
```

### CODE REVIEW

File: modules/install-roxctl-cli-linux.adoc
Comment (msugakov) [inline modules/install-roxctl-cli-linux.adoc:16]: For `ppc64le` and `s390x` Linux, `arch` will be `"-ppc64le"` and `"-s390x"` respectively.

For `x86_64` Linux, `arch` will be an empty string. This is intentional because this drives users to `/roxctl` artifacts where we did not denote architecture historically.
This can be changed to drive to `roxctl-amd64` but I don't think it's time for that yet.

**Comment by msugakov** [comment]
Oops, commands don't work in `zsh`. Let me fix this.

File: modules/install-roxctl-cli-linux.adoc
Comment (agantony) [inline modules/install-roxctl-cli-linux.adoc:12]:
```suggestion
. Download the `roxctl` CLI, which supports the `amd64`, `ppc64le`, and `s390x` architectures:
```

File: modules/install-roxctl-cli-macos.adoc
Comment (agantony) [inline modules/install-roxctl-cli-macos.adoc:12]:
```suggestion
. Download the `roxctl` CLI, which supports the `amd64` architecture:
```

File: modules/install-roxctl-cli-windows.adoc
Comment (agantony) [inline modules/install-roxctl-cli-windows.adoc:12]:
```suggestion
* Download the `roxctl` CLI, which supports the `amd64` architecture:
```

File: modules/install-roxctl-cli-linux.adoc
Comment (agantony) [inline modules/install-roxctl-cli-linux.adoc:17]:
```suggestion
[source,terminal,subs=attributes+]
----
$ arch="$(uname -m | sed "s/x86_64//")"; arch="${arch:+-$arch}"
----
+
[source,terminal,subs=attributes+]
----
$ curl -f -o roxctl "https://mirror.openshift.com/pub/rhacs/assets/{rhacs-version}/bin/Linux/roxctl${arch}"
----
```

Comment (agantony) [reply modules/install-roxctl-cli-linux.adoc:17]: I think we can break this into 2 separate code blocks for better readability

**Review by agantony — COMMENTED** [review]
Added a few suggestions; overall lgtm!

**Comment by agantony** [comment]
> Version(s): rhacs 4.3.0
> 
> **Note: I'm not sure how to mark this to be cherry-picked for `rhacs-docs-4.3` branch and whether I need to do this.**

`rhacs-docs-4.3` branch is yet to be created, so you only need to merge the PR to `rhacs-docs` as this is for 4.3.0.

> Issue: https://issues.redhat.com/browse/ROX-14701
> 
> Link to docs preview: https://65916--docspreview.netlify.app/openshift-acs/latest/cli/installing-the-roxctl-cli
> 
> QE review:
> 
> * [ ]  QE has approved this change.
> 
> Additional information: I recommend reviewing this change per commit as I tried providing meaningful commit messages in each. This complements [stackrox/stackrox#8011](https://github.com/stackrox/stackrox/pull/8011) by extending documentation.
> 
> Testing done: I created a small script with the similar content and ran it against `zsh`, `bash` and `dash` to make sure commands are valid.
> 
> ```shell
> #!/usr/bin/env zsh
> 
> set -eux
> 
> arch="$(uname -m | sed "s/x86_64//")"; arch="${arch:+-$arch}"
> #arch="$(echo "s390x" | sed "s/x86_64//")"; arch="${arch:+-$arch}"
> 
> curl -f -o roxctl "https://mirror.openshift.com/pub/rhacs/assets/4.2.0/bin/Linux/roxctl${arch}"
> ```

**Comment by agantony** [comment]
By the way, I will be creating `rhacs-docs-4.3` later this week. I would advise you to wait until you can merge this PR so that you can cherry-pick the changes to `rhacs-docs-4.3`.

**Comment by agantony** [comment]
`rhacs-docs-4.3` has been created. So you would need to merge this PR into `rhacs-docs`, and cherry-pick it into `rhacs-docs-4.3`

File: modules/install-roxctl-cli-linux.adoc
Comment (msugakov) [reply modules/install-roxctl-cli-linux.adoc]: Makes sense, done.

File: modules/install-roxctl-cli-linux.adoc
Comment (msugakov) [reply modules/install-roxctl-cli-linux.adoc:12]: I wanted to clarify about this one. Technically it is not one executable that supports three architectures. Rather, these are three different executables, one per architecture. Would it be still fair to say that "roxctl supports amd64, ppcl64le, and s390x architectures"?

**Comment by msugakov** [comment]
Thanks for your review and creating `rhacs-docs-4.3` branch @agantony 

I wasn't sure how to cherry-pick but found an [example here](https://github.com/openshift/openshift-docs/pull/65360#issuecomment-1740139072) and I think I know how to proceed now. Will do that after we converge on this change.

File: modules/install-roxctl-cli-linux.adoc
Comment (agantony) [reply modules/install-roxctl-cli-linux.adoc:12]: yeah, makes sense, perhaps we can add it as a NOTE just after the command. something like:

> . Download the `roxctl` CLI:
> <command>
> NOTE: `roxctl` CLI supports `amd64`, `ppcl64le`, and `s390x` architectures.

You can update this at all the instances.

**Comment by agantony** [comment]
https://github.com/openshift/openshift-docs/pull/65916/commits/2de7bef01f85b56225172443c38172e92791b667 LGTM

**Comment by agantony** [comment]
> Thanks for your review and creating `rhacs-docs-4.3` branch @agantony
> 
> I wasn't sure how to cherry-pick but found an [example here](https://github.com/openshift/openshift-docs/pull/65360#issuecomment-1740139072) and I think I know how to proceed now. Will do that after we converge on this change.

Regarding cherry-picking, we have a merge review team to help you out; you just need to add the `merge-review-needed` label and they will do the merging and cherry-picking operation for you, hope this helps!

File: modules/install-roxctl-cli-linux.adoc
Comment (msugakov) [reply modules/install-roxctl-cli-linux.adoc]: Moved to the `NOTE`. Yet, I keep changing the message. What do you think about the current one?

**Review by agantony — COMMENTED** [review]
Added a small suggestion.

File: modules/install-roxctl-cli-linux.adoc
Comment (msugakov) [reply modules/install-roxctl-cli-linux.adoc]: Indeed that's better. Done.

**Review by agantony — APPROVED** [review]

**Comment by msugakov** [comment]
Done: squashed to one commit.

**Pull request merged**
